scripts/dockerify.py

```diff
@@ -0,0 +1,172 @@
+#    * limitations under the License.
+
+from __future__ import print_function
+
+import argparse
+import os
+from functools import partial
+from subprocess import check_call, check_output, CalledProcessError
+from time import sleep
+
+import yaml
+
+
+EXPOSE = [22, 80, 443, 5671]
+
+# 22 isn't needed here as the user can access SSH via the internal address
+PUBLISH = ['80:80', '443:443', '5671:5671']
+
+
+def main(args=None):
+    parser = argparse.ArgumentParser(
+        description="Deploy a simple manager blueprint "
+                    "into a local docker container",
+        )
+    parser.add_argument('path', nargs='?', default='.')
+    parser.add_argument(
+        '--ssh-key',
+        default=os.path.expanduser('~/.ssh/id_rsa'),
+        )
+    parser.add_argument(
+        '--docker_tag',
+        default='cloudify/centos-manager:7',
+        )
+    parser.add_argument(
+        '--docker_context',
+        default=os.path.join(os.path.dirname(__file__), 'dockerify'),
+        )
+    parser.add_argument(
+        '--dockerfile',
+        default='https://github.com/CentOS/sig-cloud-instance-images/blob/'
+                '9681bb924c70a60dc4042dbc5fc3ed6c27aa3e1c/docker/Dockerfile',
+        )
+    args = parser.parse_args(args)
+
+    if not os.path.isfile(args.ssh_key):
+        raise ValueError(
+            "you need to create an SSH key (see man ssh-keygen) first")
+
+    id, ip = create_container(args.docker_context, args.docker_tag)
+    print("Created container: " + id)
+
+    ssh_swap(id, ip, args.ssh_key)
+
+    install(args.path, id, ip, args.ssh_key)
+
+
+def create_container(context, tag):
+    # Ensure the image is up to date
+    docker.build(['-t', tag, context])
+
+    # Create the container
+    container_id = docker.run(
+            ['--privileged', '--detach'] +
+            ['--expose={}'.format(p) for p in EXPOSE] +
+            ['--publish={}'.format(p) for p in PUBLISH] +
+            [tag],
+            ).strip()
+
+    container_ip = docker.inspect([
+        '--format={{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}',
+        container_id,
+        ]).strip()
+
+    return container_id, container_ip
+
+
+def _wait_for_file(container_id, file):
+    attempt = 0
+    while attempt < 100:
+        try:
+            return getattr(docker, 'exec')([container_id, 'cat', file])
+        except CalledProcessError as e:
+            if e.returncode != 1:
+                # Some error other than not found. Bad!
+                raise
+            attempt += 1
+            sleep(0.1)
+    raise
+
+
+def ssh_swap(id, ip, keyname):
+    """
+    You get the host key in, the client key out
+    in out in out shake it all about
+    """
+    server_pubkey_file = '/etc/ssh/ssh_host_rsa_key.pub'
+    # once this file exists the SSHD in the container is sufficiently ready
+    server_pubkey = _wait_for_file(id, server_pubkey_file)
+    with open(os.path.expanduser('~/.ssh/known_hosts'), 'a+') as f:
```

Comment:
what are we actually automatically appending this file? 
at most, i would use `ssh-keygen -R` to remove a previous entry if one already exists. All it saves is the simple step of approving a new host